onnx/common/ir.h

```diff
@@ -65,9 +68,9 @@ class ResourceGuard final {
 };
 
 struct Dimension final {
-  Dimension() : is_unknown(true) {}
-  Dimension(std::string param) : is_unknown(false), is_int(false), dim(-1), param(std::move(param)) {}
-  Dimension(int64_t dim) : is_unknown(false), is_int(true), dim(dim) {}
+  Dimension() : is_unknown(true), is_int(false), dim(-1) {}
```

Comment:
Critical bug: is_int and dim are not initialized by default, accessing them is undefined behavior before this PR